skimage/transform/histogram_matching.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+
+
+def _match_cumulative_cdf(source, template):
+    """
+    Return modified source array so that the cumulative density function of
+    its values matches the cumulative density function of the template.
+    """
+    src_values, src_unique_indices, src_counts = np.unique(source.ravel(),
+                                                           return_inverse=True,
+                                                           return_counts=True)
+    tmpl_values, tmpl_counts = np.unique(template.ravel(), return_counts=True)
+
+    # calculate normalized quantiles for each array
+    src_quantiles = np.cumsum(src_counts).astype(np.float64)
```

Comment:
Is it necessary to make the conversion explicit?

Reply:
Yes, because of the `/=` underneath. However, @sciunto, probably I would refactor as:

```python
src_quantiles = np.cumsum(src_counts) / source.size
tpl_quantiles = np.cumsum(tpl_counts) / template.size
```